### ISSUE TITLE
For/If exprs not being indented properly

### ISSUE THREAD

**Issue by XavDesbordes** [DioxusLabs/dioxus#2232]
Hi,

latest dioxus:
using rsx! format on save in the dioxus plugin for Vs code.
Linux

pictures worth 1000 words:

Normal loop in rust with my settings:
![normal_rust](https://github.com/DioxusLabs/dioxus/assets/156020158/dee7f61c-3f31-4075-98ab-86596aa344a2)

Multi scoped, normal rust (to see the behavior when an offset is applied):
![normal_rust_scoped](https://github.com/DioxusLabs/dioxus/assets/156020158/be2ce1a0-1417-402b-a969-0a66bbd60663)

inside rsx! block at top:
![inside_rsx_top](https://github.com/DioxusLabs/dioxus/assets/156020158/fdaa62d0-272d-4310-a0b5-d7a37a994a9f)

inside rsx! block:
![inside_rsx](https://github.com/DioxusLabs/dioxus/assets/156020158/2e08f7e7-832b-4c50-85ca-ea51de835bc6)

here is the loop:
```
for (my_dummy_data, txt) in data_full_profile
                                        .read()
                                        .0
                                        .as_ref()
                                        .unwrap()
                                        .get()
                                        .dummy_data_for_example[0]
                                        .dummy_section
                                        .iter()
                                        .zip(data_full_profile
                                                .read()
                                                .0
                                                .as_ref()
                                                .unwrap()
                                                .get()
                                                .dummy_data_for_example[0]
                                                .strs
                                                .iter())
                                {}
```

The wrecked indentation becomes a distraction, even more so in a longish `rsx!{}`

Here to provide further info if needed.

Thanks!

**Comment by jkelleyrtp** [comment]
Ah, it's because we don't apply indentation to formatted exprs in for/if loops - guess I didn't expect anyone to use multiline exprs in those cases! Should be simple enough to fix

**Comment by XavDesbordes** [comment]
I acknowledge being a jackass but it really eases the bulk of data I have to deal with. Do you recommend a specific strategy ? 
Preparing the iterators out of the `rsx!{}` block maybe ? (Genuinely open to suggestions to make the app the most efficient as possible)

**Comment by jkelleyrtp** [comment]
You could extract out to a vec/iterator of elements

```rust
    let profile = data_full_profile.read();
    let dummy = &profile.0.as_ref().unwrap().get().dummy_data_for_example[0];
    let profiles = dummy.dummy_section.iter().zip(dummy.strs.iter()).map(|s| {
        rsx! {
            div { "Some rendered content {s}" }
        }
    });

    rsx! {
        div { {profiles} }
    };

```

**Comment by jkelleyrtp** [comment]
We still need to format these exprs - so I'll leave this issue open and fix it in #2230

**Comment by jkelleyrtp** [comment]
Okay cool, should be released soon
